extensions/pkg/controller/backupentry/genericactuator/actuator.go

```diff
@@ -105,9 +102,35 @@ func (a *actuator) deployEtcdBackupSecret(ctx context.Context, be *extensionsv1a
 
 // Delete deletes the BackupEntry.
 func (a *actuator) Delete(ctx context.Context, be *extensionsv1alpha1.BackupEntry) error {
+	// TODO: check if we want to delete the etcd-backup secret only for the source backupentry.
+	if strings.HasPrefix(be.Name, SourceBackupNamePrefix) {
+		if err := a.deleteEtcdBackupSecret(ctx, be); err != nil {
+			return err
+		}
+	}
 	return a.backupEntryDelegate.Delete(ctx, be)
 }
 
+func (a *actuator) deleteEtcdBackupSecret(ctx context.Context, be *extensionsv1alpha1.BackupEntry) error {
+	etcdSecret := emptyEtcdBackupSecret(be.Name)
+	return kutil.DeleteObject(ctx, a.client, etcdSecret)
+}
+
+func emptyEtcdBackupSecret(backupEntryName string) *corev1.Secret {
+	secretName := BackupSecretName
+	if strings.HasPrefix(backupEntryName, SourceBackupNamePrefix) {
+		secretName = fmt.Sprintf("%s-%s", SourceBackupNamePrefix, BackupSecretName)
```

Comment:
There is already a constant for the secret name so we should use it (or, alternatively, remove the constant):
```suggestion
		secretName = SourceBackupSecretName
```